docs/installation/advanced_topics.asciidoc

```diff
@@ -107,6 +134,203 @@ View of the opened security events:
+  sql= << EOT
+  BEGIN NOT ATOMIC
+    SET @TENANT_ID = ?, @START_DATE = ?, @END_DATE = ?;
+    SELECT
+      *
+    FROM tableA
+    JOIN tableB ON tableA.id = tableB.id
+      AND date BETWEEN @START_DATE AND @END_DATE
+    WHERE tenant_id = @TENANT_ID
+      AND date BETWEEN @START_DATE AND @END_DATE
+    LIMIT ?;
+  END;
+  EOT
+  default_limit=100
+  has_date_range=enabled
+  has_limit=enabled
+
+NOTE: MySQL LIMIT will not accept a variable (`LIMIT @LIMIT`). A best practice is to always pass limit as the last field in `bindings` and to simply consume it at the end of the script (`LIMIT ?`) as shown in the last example above.
+
+==== Pagination
+
+Pagination is supported through the use of the `cursor_type`, `cursor_default`, `cursor_field`, `bindings` and `sql` attributes. Pagination supports the use of one to many columns. Special attention must be given to the order of the final result set in order to utilize the cursor properly. Symptoms of too few pages, or infinite loops through subsequent pages are signs of a mismatched cursor and/or query results order.
+
+The `limit` binding always has +1 added to it as PacketFence always consumes an extra row to determine the cursor for the following page. Due to this all conditional statements must be inclusive (ex: Bad operators "<, >", Good operators: "<=, >="). If the column value is not unique then `cursor_type=multi_field` should be used instead to avoid infinite loops.
+
+Examples of a single column cursor:
+
+  [all nodes in ascending order]
+  type=sql
+  sql= <<EOT
+    SELECT mac FROM node WHERE mac >= ? ORDER BY mac LIMIT ?;
+  EOT
+  bindings=cursor,limit
+  cursor_type=field
+  cursor_field=mac
+  default_cursor=00:00:00:00:00:00
+
+  [all nodes in descending order]
+  type=sql
+  sql= <<EOT
+    SELECT mac FROM node WHERE mac <= ? ORDER BY mac DESC LIMIT ?;
+  EOT
+  columns=mac
+  bindings=cursor,limit
+  cursor_type=field
+  cursor_field=mac
+  default_cursor=ff:ff:ff:ff:ff:ff
+
+Example of a multi column cursor:
+
+  [all ip4log logs]
+  type=sql
+  sql= <<EOT
+    SELECT
+      ip4log.ip,
+      ip4log.start_time,
+      node.mac
+    FROM ip4log
+    INNER JOIN node
+      ON ip4log.mac = node.mac
+    WHERE ip4log.start_time >= ?
+      AND node.mac >= ?
+    ORDER BY ip4log.start_time, node.mac
+    LIMIT ?;
+  EOT
+  columns=mac
+  bindings=cursor.0,cursor.1,limit
+  cursor_type=multi_field
+  cursor_field=start_time,mac
+  default_cursor=0000-00-00 00:00:00:00,00:00:00:00:00:00
+
+==== Charts
+
+Charts are defined as a comma delimited list using the `chart` attribute. An optional "@" symbol can be used to delimit a chart name. A mandatory pipe (vertical-bar) `|` is used to delimit the chart type and the fields. Within the fields a colon ":" is used to delimit each of the fields (if more than one field is necessary). The general syntax is:
+
+  charts=[pie,bar,parallel,scatter] [@ Chart Name] | field1 [:fieldN:...]
+
+There are 4 types of charts available: [options="compact"]
```

Comment:
`[options="compact"]` is an asciidoc list setting; move it on its own line, right before the list definition.